### ISSUE TITLE
[Error] No such file or directory: './snapshots/voc/siren/id-pro_maha_train.npy'

### ISSUE THREAD

**Issue by zl9501** [deeplearning-wisc/siren#3]
Hi, thanks for your amazing work!
When I run the evaluation code, there is an error ``No such file or directory: './snapshots/voc/siren/id-pro_maha_train.npy'`` about the lack of `id-pro_maha_train.npy`. In the path `./snapshots/voc/siren/`, I have the following npy files:
- `id-logits.npy`
- `id-logits_maha_train.npy`
- `id-pen.npy`
- `id-pen_maha_train.npy`
- `ood-logits.npy`
- `ood-pen.npy`

I run the code as instructed but there is no `id-pro_maha_train.npy` and I cannot find if it is generated in some middle process in the code. Thanks in advance for your kind help.

**Comment by zl9501** [comment]
I found that this error occurred because the `model` and `criterion` use the same `args` (same id), hence the sentence https://github.com/deeplearning-wisc/siren/blob/870f87bdff3269459b2975ae92815de20a681502/detr/engine.py#L223 will also set `model.args.siren` to False. 
Using a copy of `args` can solve this error.

**Comment by d12306** [comment]
@zl9501 , thanks for your suggestion! It is indeed a bug since I refactor my code for release.  I have revised it right now! Basically, resetting the flag to False is trying to avoid updating the prototypes.
